src/main/java/com/alrex/parcool/api/compatibility/WeakCache.java

```diff
@@ -0,0 +1,23 @@
+package com.alrex.parcool.api.compatibility;
+
+import java.util.Map;
+import java.util.WeakHashMap;
+
+import com.google.common.base.Supplier;
+
+public class WeakCache<K, V> {
+    private final Map<Class<?>, WeakHashMap<K, V>> cache = new WeakHashMap<>();
+
+    public V get(K key, Supplier<V> create) {
+        WeakHashMap<K,V> hashMap = cache.get(key.getClass());
+        if (hashMap == null) {
+            hashMap = new WeakHashMap<>();
+            cache.put(key.getClass(), hashMap);
```

Comment:
Why are these maps separated depending on its class of values? I think it is not necessary